src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/InitiativeFailingTab.tsx

```diff
@@ -0,0 +1,201 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useMemo } from 'react';
+import { InitiativeActionItem } from '../../../../api/types';
+import { ServiceNameAndRulesColumn } from './ServiceNameAndRulesColumn';
+import {
+  TableColumn,
+  Table as BSTable,
+  EmptyState,
+  InfoCard,
+} from '@backstage/core-components';
+import { StringIndexable } from '../../../ReportsPage/HeatmapPage/HeatmapUtils';
+import { HomepageEntity } from '../../../../api/userInsightTypes';
+import { Box, ThemeProvider, Typography } from '@material-ui/core';
+import { defaultComponentRefContext } from '../../../../utils/ComponentUtils';
+import { groupByString } from '../../../../utils/collections';
+import { humanizeAnyEntityRef } from '../../../../utils/types';
+import { LinearProgressWithLabel } from '../../../Common/LinearProgressWithLabel';
+import { percentify } from '../../../../utils/NumberUtils';
+import {
+  InitiativeFailingTabRowProps,
+  failingTabTheme,
+  useInitiativeFailingTabStyle,
+} from './InitiativeFailingTabConfig';
+
+interface InitiativeFailingTabProps {
+  actionItems: InitiativeActionItem[];
+  defaultPageSize?: number;
+  entitiesByTag: StringIndexable<HomepageEntity>;
+  numRules?: number;
+  scorecardId: number;
+}
+
+export const InitiativeFailingTab: React.FC<InitiativeFailingTabProps> = ({
+  actionItems,
+  defaultPageSize = 15,
+  entitiesByTag,
+  numRules = 2,
+  scorecardId,
+}) => {
+  const classes = useInitiativeFailingTabStyle();
+
+  const failingComponents = useMemo(
+    () => groupByString(actionItems, actionItem => actionItem.componentRef),
+    [actionItems],
+  );
+
+  const data: InitiativeFailingTabRowProps[] = useMemo(() => {
+    return Object.keys(failingComponents)
+      .map(componentRef => {
+        const tag = humanizeAnyEntityRef(
+          componentRef,
+          defaultComponentRefContext,
+        );
+
+        const { name, description } = entitiesByTag[componentRef];
+        const serviceActionItems = failingComponents[componentRef];
+
+        return {
+          actionItems: serviceActionItems,
+          componentRef,
+          description,
+          name,
+          tag,
+          score: (numRules - serviceActionItems.length) / numRules,
+        };
+      })
+      .sort((left, right) => left.actionItems.length - right.actionItems.length)
+      .sort((left, right) => left.tag.localeCompare(right.tag));
+  }, [entitiesByTag, failingComponents, numRules]);
+
+  const showPagination = useMemo(() => {
+    return Object.keys(failingComponents).length > defaultPageSize;
```

Comment:
nit: can use `_.size`
```suggestion
    return size(failingComponents) > defaultPageSize;
```

Reply:
TIL ;)